libavfilter/signature_lookup.c

```diff
@@ -244,9 +244,9 @@ static MatchingInfo* get_matching_parameters(AVFilterContext *ctx, SignatureCont
                     if (pairs[i].b[j] != pairs[k].b[l]) {
                         /* linear regression */
                         m = (pairs[k].b_pos[l]-pairs[i].b_pos[j]) / (k-i); /* good value between 0.0 - 2.0 */
-                        framerate = (int) m*30 + 0.5; /* round up to 0 - 60 */
+                        framerate = nearbyint(m*30 + 0.5); /* round up to 0 - 60 */
                         if (framerate>0 && framerate <= MAX_FRAMERATE) {
-                            offset = pairs[i].b_pos[j] - ((int) m*i + 0.5); /* only second part has to be rounded up */
+                            offset = pairs[i].b_pos[j] - nearbyint(m*i + 0.5); /* only second part has to be rounded up */
```

Comment:
```suggestion
                            offset = pairs[i].b_pos[j] - nearbyint(m*i); /* only second part has to be rounded up */
```

Reply:
Oscar noticed more accurate comparison results with the `0.5` present in there.
While it's theoretically unclear why `ceiling` performs better than `round-to-nearest-integer` we've decided to keep it as-is for now.